Calcula la frecuencia de palabras utilizando NLTK.
Treu les stopwords i realitza stemming al teu conjunt de dades.

In [34]:
import PyPDF2
from nltk.tokenize import word_tokenize
from nltk.probability import FreqDist
from nltk.corpus import stopwords
from nltk.stem import SnowballStemmer

stop_words = set(stopwords.words('english'))
stemmer = SnowballStemmer('english')

with open('Cover_letter.pdf', 'rb') as pdf_file:
    pdf_reader = PyPDF2.PdfReader(pdf_file)
    num_pages = len(pdf_reader.pages)

    text = ''
    for page_num in range(num_pages):
        page = pdf_reader.pages[page_num]
        text += page.extract_text()

tokens = word_tokenize(text.lower())

filtered_tokens = [stemmer.stem(word) for word in tokens if word.isalpha() and word not in stop_words]

freq_dist = FreqDist(filtered_tokens)

print(freq_dist.most_common(10))

[('data', 7), ('experi', 6), ('manag', 5), ('mindfuel', 4), ('skill', 4), ('team', 4), ('product', 3), ('posit', 3), ('work', 3), ('project', 3)]


Realitza sentiment analysis al teu conjunt de dades.

In [35]:
from nltk.sentiment import SentimentIntensityAnalyzer

analyzer = SentimentIntensityAnalyzer()

scores = analyzer.polarity_scores(text)

print(scores)

{'neg': 0.0, 'neu': 0.783, 'pos': 0.217, 'compound': 0.996}


La puntuación de sentimiento en mi carta de presentación indica lo siguiente:

    'neg': 0.0: No se detectó ningún sentimiento negativo en el texto.
    'neu': 0.783: El texto es principalmente neutral, lo que significa que no hay una inclinación clara hacia un sentimiento en particular.
    'pos': 0.217: Hay una pequeña inclinación positiva en el texto.
    'compound': 0.996: El puntaje compuesto es muy alto, lo que indica que en general el texto tiene una tendencia muy positiva.

En resumen, la puntuación sugiere que mi carta de presentación es principalmente neutral con una leve inclinación positiva y un tono general muy positivo.